packages/xo-web/src/xo-app/backup/health/index.js

```diff
@@ -66,26 +136,150 @@ const ACTIONS = [
   },
 ]
 
-@addSubscriptions({
-  // used by createGetLoneSnapshots
-  schedules: subscribeSchedules,
-})
-@connectStore({
-  loneSnapshots: createGetLoneSnapshots,
-  legacySnapshots: createGetObjectsOfType('VM-snapshot').filter([
-    (() => {
-      const RE = /^(?:XO_DELTA_EXPORT:|XO_DELTA_BASE_VM_SNAPSHOT_|rollingSnapshot_)/
-      return (
-        { name_label } // eslint-disable-line camelcase
-      ) => RE.test(name_label)
-    })(),
-  ]),
-  vms: createGetObjectsOfType('VM'),
-})
-export default class Health extends Component {
-  render() {
+const Health = decorate([
+  addSubscriptions({
+    // used by createGetLoneSnapshots
+    schedules: cb =>
+      subscribeSchedules(schedules => {
+        cb(keyBy(schedules, 'id'))
+      }),
+    jobs: cb =>
+      subscribeBackupNgJobs(jobs => {
+        cb(keyBy(jobs, 'id'))
+      }),
+    remotes: subscribeRemotes,
+  }),
+  connectStore({
+    loneSnapshots: createGetLoneSnapshots,
+    legacySnapshots: createGetObjectsOfType('VM-snapshot').filter([
+      (() => {
+        const RE = /^(?:XO_DELTA_EXPORT:|XO_DELTA_BASE_VM_SNAPSHOT_|rollingSnapshot_)/
+        return (
+          { name_label } // eslint-disable-line camelcase
+        ) => RE.test(name_label)
+      })(),
+    ]),
+    vms: createGetObjectsOfType('VM'),
+  }),
+  provideState({
+    initialState: () => ({
+      backupsAfterDelete: undefined,
+    }),
+    effects: {
+      async fetchBackupsAfterDelete() {
+        this.state.backupsAfterDelete = await listVmBackups(
+          toArray(this.props.remotes)
+        )
+      },
+      deleteBackups: ({ fetchBackupsAfterDelete }, backups) => {
+        const nBackups = backups.length
+        return confirm({
+          title: _('deleteBackups', { nBackups }),
+          body: _('deleteBackupsMessage', { nBackups }),
+          icon: 'delete',
+        })
+          .then(() => deleteBackups(backups))
+          .then(fetchBackupsAfterDelete, noop)
+      },
+    },
+    computed: {
+      _backupsByRemote: async (_, { remotes }) =>
+        await listVmBackups(toArray(remotes)),
+      backupsByRemote: ({ _backupsByRemote, backupsAfterDelete }) =>
+        backupsAfterDelete === undefined
+          ? _backupsByRemote
+          : backupsAfterDelete,
+      detachedBackups: ({ backupsByRemote }, { jobs, vms, schedules }) => {
```

Comment:
IMHO, you need to handle the case when `jobs`, `vms` and `schedules` are not fetched yet.  And `jobs/schedules` should be `undefined` in this case.